pcdsdevices/stopper.py

```diff
@@ -0,0 +1,137 @@
+from enum import Enum
+
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal, EpicsSignalRO
+
+from .inout import InOutPositioner, InOutPVStatePositioner
+from .interface import BaseInterface, LightpathMixin
+
+
+class Commands(Enum):
+    """Command aliases for opening and closing stoppers."""
+    close_valve = 0
+    open_valve = 1
+
+
+class Stopper(InOutPVStatePositioner):
+    """
+    Controls Stopper.
+
+    A base class for a device with two limits switches controlled via an
+    external command PV. This fully encompasses the controls Stopper
+    installations as well as un-interlocked `GateValve` s.
+
+    Parameters
+    ----------
+    prefix : str
+        Full PPS Stopper PV.
+
+    name : str
+        Alias for the stopper.
+
+    Attributes
+    ----------
+    commands : ~enum.Enum
+        An enum with integer values for `~Commands.open_valve` and
+        `~Commands.close_valve` values.
+    """
+
+    # Limit-based states
+    open_limit = Cpt(EpicsSignalRO, ':OPEN', kind='normal')
```

Comment:
If you're up for it, could be time to add `doc=` kwargs to these components, but no big deal if not.

Reply:
Sure, I'll curate the components in stopper.py